cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -4127,6 +4199,14 @@ const FunctionDoc ascii_capitalize_doc(
      "non-ASCII characters, use \"utf8_capitalize\" instead."),
     {"strings"});
 
+const FunctionDoc ascii_title_doc(
+    "Transform ASCII input to a string where the first alpha character in each word is "
+    "uppercase and all other characters are lowercase",
+    ("For each string in `strings`, return a title version.\n\n"
```

Comment:
`s/title/titlecased/`. Also add a sentence explaining better what it means, e.g.:
"Each word in the output will start with an uppercase character and its remaining characters will be lowercase."